src/Controller/CRUDController.php

```diff
@@ -61,10 +73,60 @@ public function setContainer(ContainerInterface $container = null)
         $this->configure();
     }
 
+    // BC for Symfony 3.3 where ControllerTrait exists but does not contain get() and has() methods.
+    public function __call($method, $arguments)
+    {
+        if (in_array($method, ['get', 'has'])) {
+            return call_user_func_array([$this->container, $method], $arguments);
+        }
+
+        throw new \LogicException('Call to undefined method '.__CLASS__.'::'.$method);
+    }
+
+    /**
+     * Gets a container configuration parameter by its name.
+     *
+     * @param string $name The parameter name
+     *
+     * @return mixed
+     */
+    protected function getParameter($name)
+    {
+        return $this->container->getParameter($name);
+    }
+
     /**
-     * {@inheritdoc}
+     * NEXT_MAJOR: Remove this method.
+     * @see renderWithExtraParams()
+     *
+     * @param string   $view       The view name
+     * @param array    $parameters An array of parameters to pass to the view
+     * @param Response $response   A response instance
+     *
+     * @return Response A Response instance
+     *
+     * @deprecated
      */
     public function render($view, array $parameters = [], Response $response = null)
+    {
+        @trigger_error('Method '.__CLASS__.'::render has been renamed to '.__CLASS__.'::renderWithExtraParams.'
+            .' If you need to render a template WITHOUT the additional parameters, call the templating component'
+            .' directly, such as: $this->container->get(\'twig\')->render(...);', E_USER_DEPRECATED
```

Comment:
There is no way to call `render` before your PR anyway, so I'm not sure this tip should be included.